README.adoc

```diff
@@ -1,34 +1,142 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.ConcurrentMap;
+
+@RestController
+public class CommandController {
+    @Autowired
+    private HazelcastInstance hazelcastInstance;
+
+    private ConcurrentMap<String,String> retrieveMap() {
+        return hazelcastInstance.getMap("map");
+    }
+
+    @PostMapping("/put")
+    public CommandResponse put(@RequestParam(value = "key") String key, @RequestParam(value = "value") String value) {
+        retrieveMap().put(key, value);
+        return new CommandResponse(value);
+    }
+
+    @GetMapping("/get")
+    public CommandResponse get(@RequestParam(value = "key") String key) {
+        String value = retrieveMap().get(key);
+        return new CommandResponse(value);
+    }
+}
+----
+
+== Running the Sample Application
+
+Run the application using Maven on a terminal:
+
+----
+mvn spring-boot:run -Dspring-boot.run.jvmArguments="-Dserver.port=8080"
+----
+
+Then run the application again on another terminal. But this time, note that you need to set a different value for the `server.port`.
+
+----
+mvn spring-boot:run -Dspring-boot.run.jvmArguments="-Dserver.port=8081"
+----
+
+After both application instances initialized, you should see the Hazelcast cluster formation in the output similar below:
+
+----
+Members {size:2, ver:2} [
+	Member [192.168.1.64]:5701 - 520aec3f-58a6-4fcb-a3c7-498dcf37d8ff
+	Member [192.168.1.64]:5702 - 5c03e467-d457-4847-b49a-745a335db557 this
+]
+----
+
+Now, you can make calls to application endpoints to put data and get it back. Firstly, run the following command to put the data into Hazelcast distributed map:
+
+----
+curl --data "key=key1&value=hazelcast" "localhost:8080/put"
+----
+
+You will see the value in the output. Then run the command below to get the data back. Please note that the call is made to the other application instance:
+
+----
+curl "localhost:8081/get?key=key1"
+----
+
+Again, you will see the value in the output since the data is distributed among Hazelcast cluster instances and can be accessed from any of them.
+
+== Testing the Application
+
+You can see `CommandControllerTest` test class under `src/main/test/guides/hazelcast/springboot` folder. There are three tests you can run: `testPutRequest` and `testGetRequest` use `org.springframework.test.web.reactive.server.WebTestClient` to test `/put` and `/get` REST endpoints. `testHazelcastCluster` creates another `HazelcastInstance` and checks if the cluster is formed together with the Hazelcast instance which is initialized by the Spring Boot application.
+
+To run the tests, run the following command:
+
+----
+mvn test
+----
+
+If the tests pass, you’ll see a similar output to the following:
```

Comment:
@alparslanavci I'm really sorry, but I think I was not clear here. Sorry for that.

What I meant is just that this part starting from "Now you can make calls..." should be in the section "Testing the application". I don't think we should describe the unit tests and how they are run. I think it's not interesting to the user. Also Spring Guides do not do it. Here's how I'd change it.

```suggestion
== Testing the Application

Now, you can make calls to application endpoints to put data and get it back. Firstly, run the following command to put the data into Hazelcast distributed map:

----
curl --data "key=key1&value=hazelcast" "localhost:8080/put"
----

You will see the value in the output. Then run the command below to get the data back. Please note that the call is made to the other application instance:

----
curl "localhost:8081/get?key=key1"
----

Again, you will see the value in the output since the data is distributed among Hazelcast cluster instances and can be accessed from any of them.
```

Reply:
I think this section should be called `Running the Application` and we should keep a testing section where we showcase how to run Integration Test.

Reply:
Running Application -> https://openliberty.io/guides/gradle-intro.html#running-the-application
Testing Application -> https://openliberty.io/guides/gradle-intro.html#testing-the-web-application